src/shapefile/render.rs

```diff
@@ -111,6 +118,30 @@ pub fn render(
             let (shape, record) = shape_record
                 .unwrap_or_else(|_err: shapefile::Error| (Shape::NullShape, Record::default()));
 
+
+            let mut polyline = Polyline::new(vec![Point::new(0.0, 0.0),Point::new(1.0, 1.0),]);
+            let mut polygon = Polygon::new(vec![Point::new(0.0, 0.0),Point::new(1.0, 1.0),Point::new(0.0, 0.0)].into());
+            let bbox;
+            let mut is_polyline = false;
+            if shape.shapetype() == ShapeType::Polyline {
+                is_polyline = true;
+                polyline = shapefile::Polyline::try_from(shape).unwrap();
+                bbox = polyline.bbox();
+            }
+            else if shape.shapetype() == ShapeType::Polygon {
+                polygon = Polygon::try_from(shape).unwrap();
+                bbox = polygon.bbox();
+            } else {
+                continue;
+            }
```

Comment:
I presume this logic is all to get the bounds of the shape, right? In that case there is no need to create a `mut polyline` and then reassigning immediately. I suggest something simpler (not tested):
 
```suggestion
            let (bbox, is_polyline) = if shape.shapetype() == ShapeType::Polyline {
                let polyline = shapefile::Polyline::try_from(shape).unwrap();
                (polyline.bbox(), true)
            } else if shape.shapetype() == ShapeType::Polygon {
                let polygon = Polygon::try_from(shape).unwrap();
                (polygon.bbox(), false)
            } else {
                continue;
            };
```